spec/config_spec.rb

```diff
@@ -665,6 +665,136 @@ def self.name
       end
     end
 
+    context "with variety of required attributes combinations with env option" do
+      before { allow(Anyway::Settings).to receive(:current_environment).and_return("production") }
+
+      let(:app_config) do
+        Class.new(described_class) do
+          config_name "cool_app"
+
+          attr_config :host, :port, :api_key, :user, :password, :debug, :redis_host
+          required :host, :port
+          required :api_key, env: "production"
+          required :user, :password, env: %w[development production]
+          required :redis_host, env: {except: :test}
+        end
+      end
+
+      let(:demo_config) do
+        Class.new(described_class) do
+          config_name "demo"
+          attr_config :sentry_api_key
+
+          required :sentry_api_key, env: {except: %i[local development test]}
+        end
+      end
+
+      let(:missed_keys) { [] }
+      let(:config_values) do
+        {
+          api_key: "123",
+          user: "john",
+          password: "simple",
+          redis_host: "localhost:6379",
+          host: "localhost",
+          port: 80
+        }.reject { |k| missed_keys.include?(k) }
+      end
+      let(:error_msg) { /missing or empty: #{missed_keys.join(', ')}$/ }
+
+      subject { app_config.new(config_values) }
+
+      it "not to raise ValidationError when all values are presence" do
+        expect { subject }.to_not raise_error(Anyway::Config::ValidationError)
```

Comment:
@fargelus FYI

> WARNING: Using `expect { }.not_to raise_error(SpecificErrorClass, message)` risks false positives, since literally any other error would cause the expectation to pass, including those raised by Ruby (e.g. `NoMethodError`, `NameError` and `ArgumentError`)...